MediaBrowser.Controller/Entities/BaseItem.cs

```diff
@@ -1408,6 +1408,22 @@ public string GetPresentationUniqueKey()
 
         public virtual bool RequiresRefresh()
         {
+            if (string.IsNullOrEmpty(Path) || DateModified == default)
+            {
+                return false;
+            }
+
+            var info = FileSystem.GetFileSystemInfo(Path);
+            if (info.Exists)
+            {
+                if (info.IsDirectory)
+                {
+                    return info.LastWriteTimeUtc != DateModified;
+                }
+
+                return info.LastWriteTimeUtc != DateModified && info.Length != (Size ?? 0);
```

Comment:
Can we be sure that LastWriteTime is always "correct" i.e. will all filesystems report it?

Reply:
We checked it on NTFS, ext4, ZFS, samba and NFS and it worked correctly on all of them